backend/server/core/services/getAchievements.service.ts

```diff
@@ -11,29 +11,28 @@ import {Task} from "../interfaces/task.interface";
 // type GetAchievements = (challengeId: number) => ActualAchievement[];
 
 export function getAchievements(
-    challengeId: number,
-    challenges: Challenge[]
+  challengeId: number,
+  challenges: Challenge[]
 ): ActualAchievement[] {
-    const challenge: Challenge = challenges.find(
-        (challenge) => challenge.id === challengeId
-    );
+  const challenge: Challenge = challenges.find(
+    (challenge) => challenge.id === challengeId
+  );
 
-    const timeBetween = new Date().getTime() - challenge.startDate.getTime();
-    const daysBetween = Math.floor(timeBetween / (1000 * 3600 * 24));
-    const passedTasks: Task[] = challenge.tasksOrder.slice(0, daysBetween);
+  const timeBetween = new Date().getTime() - challenge.startDate.getTime();
+  const daysBetween = Math.floor(timeBetween / (1000 * 3600 * 24));
+  const passedTasks: Task[] = challenge.tasksOrder.slice(0, daysBetween);
 
-    const achievements: ActualAchievement[] = passedTasks.map(passedTask => {
-            const taskStatus: Status = {
-                state: StateItem.SUCCESS,
-                updated: new Date(),
-            };
-            const actualAchievement: ActualAchievement = {
-                id: passedTask.id,
-                description: passedTask.description,
-                status: taskStatus,
-            };
-            return actualAchievement;
-        }
-    );
-    return achievements;
+  const achievements: ActualAchievement[] = passedTasks.map((passedTask) => {
+    const taskStatus: Status = {
```

Comment:
It means here we need to find a challenge by `id` firstly. Then, as every challenge must have several achievements (consider challenge like object, which has property `achievements:  Array<string>`), we need just to return achievements from it.